pkgs/development/compilers/swift/default.nix

```diff
@@ -0,0 +1,251 @@
+      "1gjcr6g3ffs3nhf4a84iwg4flbd7rqcf9rvvclwyq96msa3mj950";
+    cmark = fetch
+      "swift-cmark"
+      "0qf2f3zd8lndkfbxbz6vkznzz8rvq5gigijh7pgmfx9fi4zcssqx";
+    lldb = fetch
+      "swift-lldb"
+      "023ikrahs5ly27cm1f4s4s8plgcglmv3vq848jcd5dmfyijgzxnb";
+    llbuild = fetch
+      "swift-llbuild"
+      "01bz8cbshfcf6ph3a5x808azi2fabjanjr8838jcldhbl2q7wx7b";
+    pm = fetch
+      "swift-package-manager"
+      "1bgzv8xcm1jc754h3hzi42ax6brqikr8k25n5c6vjy76iw9zgbcf";
+    xctest = fetch
+      "swift-corelibs-xctest"
+      "0cj5y7wanllfldag08ci567x12aw793c79afckpbsiaxmwy4xhnm";
+    foundation = fetch
+      "swift-corelibs-foundation"
+      "0dds1h2k97mxs6v5g5wd42yf3xpp075j95cygxh2lwirwnianqz3";
+    libdispatch = fetch
+      "swift-corelibs-libdispatch"
+      "1lq534nr72d1q4l5hzcpd8l586xdzxi97l4lyjdyl1ri05wylkc9";
+    swift = fetch
+      "swift"
+      "1cg0b5mglmgp9wmd19w0nxigz1jgfnj3ymfm97zzawg07jx8b46n";
+  };
+
+  devInputs = [
+    curl
+    glibc
+    icu
+    libblocksruntime
+    libbsd
+    libedit
+    libuuid
+    libxml2
+    ncurses
+    sqlite
+    swig
+    #    systemtap?
+  ];
+
+  cmakeFlags = [
+    "-DGLIBC_INCLUDE_PATH=${stdenv.cc.libc.dev}/include"
+    "-DC_INCLUDE_DIRS=${stdenv.lib.makeSearchPathOutput "dev" "include" devInputs}:${libxml2.dev}/include/libxml2"
+    "-DGCC_INSTALL_PREFIX=${clang.cc.gcc}"
+  ];
+
+  builder = ''
+    $SWIFT_SOURCE_ROOT/swift/utils/build-script \
+      --preset-file=${./build-presets.ini} \
+      --preset=buildbot_linux \
+      installable_package=$INSTALLABLE_PACKAGE \
+      install_prefix=$out \
+      install_destdir=$SWIFT_INSTALL_DIR \
+      extra_cmake_options="${stdenv.lib.concatStringsSep "," cmakeFlags}"'';
+
+in
+stdenv.mkDerivation rec {
+  name = "swift-${version_friendly}";
+
+  buildInputs = devInputs ++ [
+    autoconf
+    automake
+    bash
+    clang
+    cmake
+    coreutils
+    libtool
+    ninja
+    perl
+    pkgconfig
+    python
+    rsync
+    which
+    paxctl
+    findutils
+  ];
+
+  # TODO: Revisit what's propagated and how
+  propagatedBuildInputs = [
+    libgit2
+    python
+  ];
+  propagatedUserEnvPkgs = [ git pkgconfig ];
+
+  hardeningDisable = [ "format" ]; # for LLDB
+
+  configurePhase = ''
+    cd ..
+    
+    export INSTALLABLE_PACKAGE=$PWD/swift.tar.gz
+
+    mkdir build install
+    export SWIFT_BUILD_ROOT=$PWD/build
+    export SWIFT_INSTALL_DIR=$PWD/install
+
+    cd $SWIFT_BUILD_ROOT
+
+    unset CC
+    unset CXX
+
+    export NIX_ENFORCE_PURITY=
+  '';
+
+  unpackPhase = ''
+    mkdir src
+    cd src
+    export sourceRoot=$PWD
+    export SWIFT_SOURCE_ROOT=$PWD
+
+    cp -r ${sources.clang} clang
+    cp -r ${sources.llvm} llvm
+    cp -r ${sources.compilerrt} compiler-rt
+    cp -r ${sources.cmark} cmark
+    cp -r ${sources.lldb} lldb
+    cp -r ${sources.llbuild} llbuild
+    cp -r ${sources.pm} swiftpm
+    cp -r ${sources.xctest} swift-corelibs-xctest
+    cp -r ${sources.foundation} swift-corelibs-foundation
```

Comment:
Yep! What's built is described by the preset we're using, which is a modified version of what they use on the linux buildbots:
https://github.com/NixOS/nixpkgs/pull/22098/files#diff-207758b47bdae6fde307317858e26f47R715
(which references the `mixin_linux_installation` preset: https://github.com/NixOS/nixpkgs/pull/22098/files#diff-207758b47bdae6fde307317858e26f47R681).

It's possible we can break these apart eventually but I don't think it would be easy or even worth it--AFAICT the components are all very much meant to be built/tested together.

Also this does take a while to build... :).